pennylane/tape/tapes/qubit_param_shift.py

```diff
@@ -354,6 +360,12 @@ def processing_fn(results):
                 array[float]: 1-dimensional array of length determined by the tape output
                 measurement statistics
             """
-            return np.dot(coeffs, np.squeeze(results))
+            results = np.squeeze(results)
+
+            if results.dtype is not np.dtype("O"):
+                dot = lambda x: np.dot(coeffs, x)
+                return np.apply_along_axis(dot, 0, results)
+
+            return sum([c * r for c, r in zip(coeffs, results)])
 

```

Comment:
Is it suspicious that this line is not covered, considering that this should be the usual case?

Reply:
Yep! I can add a test for
```python
return qml.var(qml.PauliZ(0)), qml.probs([1, 2])
```

Reply:
Have added in f32b68f